--- a/src/freedreno/vulkan/tu_device.c
+++ b/src/freedreno/vulkan/tu_device.c
@@ -28,7 +28,7 @@
 #include "tu_pass.h"
 #include "tu_pipeline.h"
 
-#define TU_DEVICE_BINARY_CACHE_SIZE 64
+#define TU_DEVICE_BINARY_CACHE_SIZE 1024
 
 static int
 tu_get_debug_flags()
@@ -520,7 +520,7 @@
    device->physical_device = physical_device;
    device->instance = physical_device->instance;
-   device->num_priority_levels = 2;
+   device->num_priority_levels = 8;
 
    tu_device_get_cache_uuid(device->cache_uuid);
 
--- a/src/freedreno/vulkan/tu_cmd_buffer.c
+++ b/src/freedreno/vulkan/tu_cmd_buffer.c
@@ -820,6 +820,9 @@
    if (cmd->state.pass)
       tu_emit_cache_flush(cmd);
 
+   /* Florkaa Turbo: Aggressive LRZ and GMEM */
+   cmd->state.lrz.fast_clear = true;
+   cmd->state.lrz.gpu_dir_write = true;
+
    tu_cs_emit_call(cs, &cmd->state.pass->gmem_config);
 }
 
--- a/src/freedreno/ir3/ir3_compiler.c
+++ b/src/freedreno/ir3/ir3_compiler.c
@@ -110,7 +110,7 @@
    compiler->gpu_id = gpu_id;
    compiler->options = options;
-   compiler->max_regs = 128;
+   compiler->max_regs = 72;
+   compiler->opt_aggressive = true;
 
    if (compiler->gpu_id >= 600) {
       compiler->branch_dest_alignment = 4;
