--- a/src/freedreno/vulkan/tu_device.c
+++ b/src/freedreno/vulkan/tu_device.c
@@ -28,7 +28,7 @@
 #include "tu_pass.h"
 #include "tu_pipeline.h"
 
-#define TU_DEVICE_BINARY_CACHE_SIZE 64
+#define TU_DEVICE_BINARY_CACHE_SIZE 1024
 
 static int
 tu_get_debug_flags()
--- a/src/freedreno/ir3/ir3_compiler.c
+++ b/src/freedreno/ir3/ir3_compiler.c
@@ -110,7 +110,7 @@
    compiler->gpu_id = gpu_id;
    compiler->options = options;
-   compiler->max_regs = 128;
+   compiler->max_regs = 96; /* SD870 Stability Spot */
 
    if (compiler->gpu_id >= 600) {
       compiler->branch_dest_alignment = 4;
--- a/src/freedreno/common/freedreno_devices.py
+++ b/src/freedreno/common/freedreno_devices.py
@@ -100,2 +100,18 @@
-a6xx_base = GPUProps(
-    reg_size_vec4 = 128,
+a6xx_base = GPUProps(
+    reg_size_vec4 = 128,
+    sysmem_vpc_attr_buf_size = 131072,
+    sysmem_vpc_pos_buf_size = 65536,
+    sysmem_vpc_bv_pos_buf_size = 32768,
+    sysmem_ccu_color_cache_fraction = 1.0,
+    sysmem_per_ccu_color_cache_size = 128 * 1024,
+    sysmem_ccu_depth_cache_fraction = 1.0,
+    sysmem_per_ccu_depth_cache_size = 256 * 1024,
+    gmem_vpc_attr_buf_size = 49152,
+    gmem_vpc_pos_buf_size = 24576,
+    gmem_vpc_bv_pos_buf_size = 32768,
+    gmem_ccu_color_cache_fraction = 1.0,
+    gmem_per_ccu_color_cache_size = 128 * 1024,
+    gmem_ccu_depth_cache_fraction = 1.0,
+    gmem_per_ccu_depth_cache_size = 256 * 1024,
+)
